[2025-03-06T16:13:54.884+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-03-06T16:13:54.912+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_flow.load_exchangeRate_python_job manual__2025-03-06T16:13:51.096323+00:00 [queued]>
[2025-03-06T16:13:54.924+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_flow.load_exchangeRate_python_job manual__2025-03-06T16:13:51.096323+00:00 [queued]>
[2025-03-06T16:13:54.927+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 1
[2025-03-06T16:13:54.941+0000] {taskinstance.py:2890} INFO - Executing <Task(SparkSubmitOperator): load_exchangeRate_python_job> on 2025-03-06 16:13:51.096323+00:00
[2025-03-06T16:13:54.948+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'spark_flow', 'load_exchangeRate_python_job', 'manual__2025-03-06T16:13:51.096323+00:00', '--job-id', '15', '--raw', '--subdir', 'DAGS_FOLDER/dag_***_spark.py', '--cfg-path', '/tmp/tmpuo8_m1u_']
[2025-03-06T16:13:54.951+0000] {standard_task_runner.py:105} INFO - Job 15: Subtask load_exchangeRate_python_job
[2025-03-06T16:13:54.953+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=816) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-03-06T16:13:54.954+0000] {standard_task_runner.py:72} INFO - Started process 818 to run task
[2025-03-06T16:13:55.013+0000] {task_command.py:467} INFO - Running <TaskInstance: spark_flow.load_exchangeRate_python_job manual__2025-03-06T16:13:51.096323+00:00 [running]> on host 6dfac24f2f32
[2025-03-06T16:13:55.116+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='haonv' AIRFLOW_CTX_DAG_ID='spark_flow' AIRFLOW_CTX_TASK_ID='load_exchangeRate_python_job' AIRFLOW_CTX_EXECUTION_DATE='2025-03-06T16:13:51.096323+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-03-06T16:13:51.096323+00:00'
[2025-03-06T16:13:55.117+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-03-06T16:13:55.146+0000] {base.py:84} INFO - Retrieving connection 'spark-conn'
[2025-03-06T16:13:55.147+0000] {spark_submit.py:474} INFO - Spark-Submit cmd: spark-submit --master spark://172.18.0.4:7077 --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py
[2025-03-06T16:13:59.106+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO SparkContext: Running Spark version 3.5.5
[2025-03-06T16:13:59.109+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO SparkContext: OS info Linux, 5.15.167.4-microsoft-standard-WSL2, amd64
[2025-03-06T16:13:59.110+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO SparkContext: Java version 17.0.14
[2025-03-06T16:13:59.162+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-03-06T16:13:59.267+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO ResourceUtils: ==============================================================
[2025-03-06T16:13:59.268+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-03-06T16:13:59.269+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO ResourceUtils: ==============================================================
[2025-03-06T16:13:59.269+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO SparkContext: Submitted application: ExchangeRateToDW
[2025-03-06T16:13:59.294+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-03-06T16:13:59.304+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO ResourceProfile: Limiting resource is cpu
[2025-03-06T16:13:59.305+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-03-06T16:13:59.360+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO SecurityManager: Changing view acls to: ***
[2025-03-06T16:13:59.361+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO SecurityManager: Changing modify acls to: ***
[2025-03-06T16:13:59.362+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO SecurityManager: Changing view acls groups to:
[2025-03-06T16:13:59.362+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO SecurityManager: Changing modify acls groups to:
[2025-03-06T16:13:59.363+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2025-03-06T16:13:59.671+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO Utils: Successfully started service 'sparkDriver' on port 41081.
[2025-03-06T16:13:59.705+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO SparkEnv: Registering MapOutputTracker
[2025-03-06T16:13:59.746+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO SparkEnv: Registering BlockManagerMaster
[2025-03-06T16:13:59.765+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-03-06T16:13:59.766+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-03-06T16:13:59.772+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-03-06T16:13:59.798+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-113ba8ec-16b4-4dc2-bfb1-5ed17eebeede
[2025-03-06T16:13:59.815+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2025-03-06T16:13:59.839+0000] {spark_submit.py:641} INFO - 25/03/06 16:13:59 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-03-06T16:14:00.082+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:00 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2025-03-06T16:14:00.167+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:00 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2025-03-06T16:14:00.208+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:00 ERROR SparkContext: Failed to add /opt/bitnami/spark/jars/postgresql-42.7.5.jar to Spark environment
[2025-03-06T16:14:00.208+0000] {spark_submit.py:641} INFO - java.io.FileNotFoundException: Jar /opt/bitnami/spark/jars/postgresql-42.7.5.jar not found
[2025-03-06T16:14:00.209+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.addLocalJarFile$1(SparkContext.scala:2095)
[2025-03-06T16:14:00.210+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.addJar(SparkContext.scala:2151)
[2025-03-06T16:14:00.211+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.$anonfun$new$15(SparkContext.scala:521)
[2025-03-06T16:14:00.212+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.$anonfun$new$15$adapted(SparkContext.scala:521)
[2025-03-06T16:14:00.213+0000] {spark_submit.py:641} INFO - at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2025-03-06T16:14:00.213+0000] {spark_submit.py:641} INFO - at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2025-03-06T16:14:00.214+0000] {spark_submit.py:641} INFO - at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2025-03-06T16:14:00.215+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.<init>(SparkContext.scala:521)
[2025-03-06T16:14:00.216+0000] {spark_submit.py:641} INFO - at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
[2025-03-06T16:14:00.217+0000] {spark_submit.py:641} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2025-03-06T16:14:00.218+0000] {spark_submit.py:641} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2025-03-06T16:14:00.219+0000] {spark_submit.py:641} INFO - at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2025-03-06T16:14:00.219+0000] {spark_submit.py:641} INFO - at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2025-03-06T16:14:00.220+0000] {spark_submit.py:641} INFO - at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2025-03-06T16:14:00.221+0000] {spark_submit.py:641} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2025-03-06T16:14:00.223+0000] {spark_submit.py:641} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2025-03-06T16:14:00.223+0000] {spark_submit.py:641} INFO - at py4j.Gateway.invoke(Gateway.java:238)
[2025-03-06T16:14:00.224+0000] {spark_submit.py:641} INFO - at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2025-03-06T16:14:00.225+0000] {spark_submit.py:641} INFO - at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2025-03-06T16:14:00.226+0000] {spark_submit.py:641} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-03-06T16:14:00.227+0000] {spark_submit.py:641} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-03-06T16:14:00.227+0000] {spark_submit.py:641} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2025-03-06T16:14:00.331+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:00 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://172.18.0.4:7077...
[2025-03-06T16:14:00.380+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:00 INFO TransportClientFactory: Successfully created connection to /172.18.0.4:7077 after 27 ms (0 ms spent in bootstraps)
[2025-03-06T16:14:00.482+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:00 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20250306161400-0004
[2025-03-06T16:14:00.493+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:00 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 43239.
[2025-03-06T16:14:00.494+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:00 INFO NettyBlockTransferService: Server created on 6dfac24f2f32:43239
[2025-03-06T16:14:00.497+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:00 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-03-06T16:14:00.506+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:00 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 6dfac24f2f32, 43239, None)
[2025-03-06T16:14:00.512+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:00 INFO BlockManagerMasterEndpoint: Registering block manager 6dfac24f2f32:43239 with 434.4 MiB RAM, BlockManagerId(driver, 6dfac24f2f32, 43239, None)
[2025-03-06T16:14:00.515+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:00 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 6dfac24f2f32, 43239, None)
[2025-03-06T16:14:00.517+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:00 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 6dfac24f2f32, 43239, None)
[2025-03-06T16:14:00.871+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:00 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2025-03-06T16:14:01.314+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:01 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2025-03-06T16:14:01.331+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:01 INFO SharedState: Warehouse path is 'file:/opt/***/spark-warehouse'.
[2025-03-06T16:14:03.403+0000] {spark_submit.py:641} INFO - Traceback (most recent call last):
[2025-03-06T16:14:03.405+0000] {spark_submit.py:641} INFO - File "/opt/***/jobs/python/load_exchangeRateToDW.py", line 66, in <module>
[2025-03-06T16:14:03.410+0000] {spark_submit.py:641} INFO - .load()
[2025-03-06T16:14:03.411+0000] {spark_submit.py:641} INFO - ^^^^^^
[2025-03-06T16:14:03.412+0000] {spark_submit.py:641} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/streaming/readwriter.py", line 304, in load
[2025-03-06T16:14:03.413+0000] {spark_submit.py:641} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2025-03-06T16:14:03.414+0000] {spark_submit.py:641} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 185, in deco
[2025-03-06T16:14:03.440+0000] {spark_submit.py:641} INFO - pyspark.errors.exceptions.captured.AnalysisException: Failed to find data source: kafka. Please deploy the application as per the deployment section of Structured Streaming + Kafka Integration Guide.
[2025-03-06T16:14:03.576+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:03 INFO SparkContext: Invoking stop() from shutdown hook
[2025-03-06T16:14:03.577+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:03 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2025-03-06T16:14:03.610+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:03 INFO SparkUI: Stopped Spark web UI at http://6dfac24f2f32:4040
[2025-03-06T16:14:03.630+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:03 INFO StandaloneSchedulerBackend: Shutting down all executors
[2025-03-06T16:14:03.632+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:03 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Asking each executor to shut down
[2025-03-06T16:14:03.640+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:03 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20250306161400-0004/0 on worker-20250306160751-172.18.0.5-46537 (172.18.0.5:46537) with 2 core(s)
[2025-03-06T16:14:03.645+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:03 INFO StandaloneSchedulerBackend: Granted executor ID app-20250306161400-0004/0 on hostPort 172.18.0.5:46537 with 2 core(s), 1024.0 MiB RAM
[2025-03-06T16:14:03.647+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:03 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20250306161400-0004/1 on worker-20250306160752-172.18.0.6-39231 (172.18.0.6:39231) with 2 core(s)
[2025-03-06T16:14:03.648+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:03 INFO StandaloneSchedulerBackend: Granted executor ID app-20250306161400-0004/1 on hostPort 172.18.0.6:39231 with 2 core(s), 1024.0 MiB RAM
[2025-03-06T16:14:03.670+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:03 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2025-03-06T16:14:03.702+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:03 INFO MemoryStore: MemoryStore cleared
[2025-03-06T16:14:03.702+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:03 INFO BlockManager: BlockManager stopped
[2025-03-06T16:14:03.710+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:03 INFO BlockManagerMaster: BlockManagerMaster stopped
[2025-03-06T16:14:03.713+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:03 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2025-03-06T16:14:03.728+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:03 INFO SparkContext: Successfully stopped SparkContext
[2025-03-06T16:14:03.729+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:03 INFO ShutdownHookManager: Shutdown hook called
[2025-03-06T16:14:03.729+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:03 INFO ShutdownHookManager: Deleting directory /tmp/spark-2055b92f-2238-4dbf-8eb8-c22f09d69d8a
[2025-03-06T16:14:03.734+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:03 INFO ShutdownHookManager: Deleting directory /tmp/spark-9281a839-2d75-49b8-88fa-56da78272907
[2025-03-06T16:14:03.738+0000] {spark_submit.py:641} INFO - 25/03/06 16:14:03 INFO ShutdownHookManager: Deleting directory /tmp/spark-2055b92f-2238-4dbf-8eb8-c22f09d69d8a/pyspark-475cce5d-ad40-47f4-b771-dd03aef351c7
[2025-03-06T16:14:03.788+0000] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 177, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 567, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://172.18.0.4:7077 --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py. Error code is: 1.
[2025-03-06T16:14:03.792+0000] {taskinstance.py:1226} INFO - Marking task as FAILED. dag_id=spark_flow, task_id=load_exchangeRate_python_job, run_id=manual__2025-03-06T16:13:51.096323+00:00, execution_date=20250306T161351, start_date=20250306T161354, end_date=20250306T161403
[2025-03-06T16:14:03.826+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-03-06T16:14:03.826+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 15 for task load_exchangeRate_python_job (Cannot execute: spark-submit --master spark://172.18.0.4:7077 --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py. Error code is: 1.; 818)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 177, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 567, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://172.18.0.4:7077 --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py. Error code is: 1.
[2025-03-06T16:14:03.846+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-03-06T16:14:03.873+0000] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-03-06T16:14:03.879+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
